File: Brute-force/recursive_staircase.py
# implement recursive staircase
def recursive_staircase(n):
    if n == 1 or n == 0:
        return 1
    elif n == 2
        return  2
    else:
        return recursive_staircase(n-3)+recursive_staircase(n-2)+recursive_staircase(n-1)
# driver function
if __name__ == '__main__':
    n = int(input("enter number of stairs"))
    print("no. of solutions", recursive_staircase(n))

File: Brute Force/RecursiveStaircase.scala

/*Program in Scala for Recursive Staircase*/

object RecursiveStaircase extends App {
    //Function for starcase
  def recStaircase(stairCase: Int): Int =
  {
      //multiple match cases for different conditions
        stairCase match
        {
            case 0 => 0
            case 1 => 1
            case 2 => 2
            //default case
            case _ => recStaircase(stairCase-1) + recStaircase(stairCase-2)
        }
  }
  val stairs = 8
  //displaying the output
  println("Total number of ways are : "+recStaircase(stairs))
}